hazelcast/src/main/java/com/hazelcast/internal/server/tcp/TcpServer.java

```diff
@@ -233,48 +182,87 @@ public synchronized void shutdown() {
+    public Collection<ServerConnection> getActiveConnections() {
+        if (unifiedConnectionManager != null) {
+            return unifiedConnectionManager.getActiveConnections();
+        }
+
+        Set<ServerConnection> connections = null;
+        for (TcpServerConnectionManager connectionManager : connectionManagers.values()) {
+            Collection<ServerConnection> endpointConnections = connectionManager.getActiveConnections();
+            if (endpointConnections != null && !endpointConnections.isEmpty()) {
+                if (connections == null) {
+                    connections = new HashSet<>();
+                }
+
+                connections.addAll(endpointConnections);
+            }
+        }
+        return connections == null ? Collections.emptySet() : connections;
     }
 
-    /**
-     * Returns the respective endpoint manager based on the qualifier.
-     * Under unified endpoint environments, this will return the respective view of the {@link UnifiedServerConnectionManager}
-     * eg. {@link MemberViewUnifiedServerConnectionManager} or {@link ClientViewUnifiedEndpointManager} which report
-     * connections based on the qualifier, but they register/create connection directly on the Unified manager.
-     *
-     * @param qualifier
-     * @return
-     */
-    public ServerConnectionManager getConnectionManager(EndpointQualifier qualifier) {
-        ServerConnectionManager mgr = connectionManagers.get(qualifier);
-        if (mgr == null) {
-            logger.finest("An connection manager for qualifier " + qualifier + " was never registered.");
+    @Override
+    public Map<EndpointQualifier, NetworkStats> getNetworkStats() {
+        if (unifiedConnectionManager != null) {
+            return emptyMap();
         }
 
-        return mgr;
+        Map<EndpointQualifier, NetworkStats> stats = null;
+        for (Map.Entry<EndpointQualifier, TcpServerConnectionManager> entry : connectionManagers.entrySet()) {
+            if (stats == null) {
+                stats = new HashMap<>();
+            }
+            stats.put(entry.getKey(), entry.getValue().getNetworkStats());
+        }
+        return stats == null ? emptyMap() : stats;
     }
 
-    ServerConnectionManager getUnifiedOrDedicatedEndpointManager(EndpointQualifier qualifier) {
-        return unifiedConnectionManager != null ? unifiedConnectionManager : connectionManagers.get(qualifier);
+    @Override
+    public void addConnectionListener(ConnectionListener<ServerConnection> listener) {
+        if (unifiedConnectionManager != null) {
+            unifiedConnectionManager.addConnectionListener(listener);
+        } else {
+            connectionManagers.values().forEach(manager -> manager.addConnectionListener(listener));
+        }
+    }
+
+    @Override
+    public TcpServerConnectionManager getConnectionManager(EndpointQualifier qualifier) {
+        if (unifiedConnectionManager != null) {
+            return unifiedConnectionManager;
```

Comment:
The purpose of the `[Client|Member|...]ViewUnifiedConnectionManager`s was to present the corresponding view of the unified connection manager with the `ConnectionManager` API. So client code using `getConnectionManager(CLIENT)` would get a `ConnectionManager` presenting a view of `CLIENT` connections backed by the unified connection manager.

Assuming 2 members with single-socket configuration (`advancedNetworkConfig#enabled == false`) + 1 client connected, then
`member1Node.getServer().getConnectionManager(CLIENT).getActiveConnections();` previously returned just 1 connection with `CLIENT` protocol type. With this PR, this will return the `unifiedConnectionManager` itself -> `getActiveConnections()` will return 2 connections (1 to the other cluster member + 1 to the client) which does not make sense from the point of view of an API client requesting `getConnectionManager(CLIENT)`.

Reply:
```
The purpose of the [Client|Member|...]ViewUnifiedConnectionManagers was to present the corresponding view of the unified connection manager with the ConnectionManager API. 
```

It might be the goal, but the actual implementation isn't a view because proper filtering isn't done. For example the MemberViewUnifiedServerConnectionManager will not return only members:
```
 @Override
    public Collection<ServerConnection> getConnections() {
        return unifiedEndpointManager.getConnections();
    }
```
It will also return clients. 

And if you would register for example a connection listener on the MemberViewUnifiedServerConnectionManager and a client connection gets added, the listener will get triggered on this member view.

The view approach is broken and causes a huge amount of complexity.

Reply:
As discussed, the `getConnectionManager(EndpointQualifier)` API purpose is now clarified in javadoc. Wrong stats due to missing connection filtering will be addressed in a further PR, along with cleanup of `getConnections` / `getActiveConnections`